src/libplctag/Tag.cs

```diff
@@ -168,6 +177,15 @@ public void Write(int millisecondTimeout)
 
         }
 
+        public async Task WriteAsync(int millisecondTimeout, CancellationToken token = default)
+        {
+            using (var tcs = CancellationTokenSource.CreateLinkedTokenSource(token))
```

Comment:
Should this and the one above be `cts`?
Also, what does `CreateLinkedTokenSource` do with a `null`? It seems like that might throw.

Reply:
Nice catch(es)!

I didn't know this till you asked the question, but the default CancellationToken is CancellationToken.None which works OK with `CreateLinkedTokenSource`